File: Bronze/567-FencePainting/567.py
import sys

sys.stdin = open("paint.in", "r")
sys.stdout = open("paint.out", "w")

john = [int(x) for x in input().split(" ")]
bessie = [int(x) for x in input().split(" ")]

if bessie[0] < john[0]:
    if bessie[1] < john[0]:
        print(bessie[1] - bessie[0] + john[1] - john[0])
    elif bessie[1] >= john[1]:
        print(bessie[1] - bessie[0])
    elif bessie[1] >= john[0] and bessie[1] < john[1]:
        print(john[1] - bessie[0])
elif bessie[0] >= john[1]:
    print(bessie[1] - bessie[0] + john[1] - john[0])
elif bessie[0] >= john[0] and bessie[0] < john[1]:
    if bessie[1] < john[1]:
        print(john[1] - john[0])
    elif bessie[1] >= john[1]:
        print(bessie[1] - john[0])

File: Bronze/567-FencePainting/567.cpp
#include <iostream>
#include <cstdio>

using namespace std;

void setIO(string s) {
  freopen((s + ".in").c_str(), "r", stdin);
  freopen((s + ".out").c_str(), "w", stdout);
}

int main() {
  setIO("paint");
  
  int a;
  int b;
  int c;
  int d;

  cin >> a >> b;
  cin >> c >> d;

  if (c < a) {
    if (d < a) {
      cout << d - c + b - a << "\n";
    } else if (d >= b) {
      cout << d - c << "\n";
    } else if (d >= a && d < b) {
      cout << b - c << "\n";
    }
  } else if (c >= b) {
    cout << d - c + b - a << "\n";
  } else if (c >= a && c < b) {
    if (d < b) {
      cout << b - a << "\n";
    } else if (d >= b) {
      cout << d - a << "\n";
    }
  }
}

